src/registry.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/* eslint-disable no-underscore-dangle */
+
+import { URL } from 'url';
+import * as path from 'path';
+import * as os from 'os';
+import { exec } from 'shelljs';
+import { fs, SfdxError } from '@salesforce/core';
+import { Env } from '@salesforce/kit';
+
+export class Registry {
+  public get registryEntryLocal(): string {
+    return this._registryEntryLocal;
+  }
+
+  public get registryEntryGlobal(): string {
+    return this._registryEntryGlobal;
+  }
+
+  public get registryUrl(): string {
+    return this._registryUrl;
+  }
+
+  public get authToken(): string {
+    return this._authToken;
+  }
+  private _registryEntryLocal: string;
+  private _registryEntryGlobal: string;
+  private env: Env;
+
+  public constructor(private _registryUrl?, private _authToken?: string) {
+    this.init();
+    this.loadNpmConfigs();
+  }
+
+  /**
+   * Return a properly formatted --registry string
+   */
+  public getRegistryParameter(): string {
+    return `--registry ${this.registryUrl}`;
+  }
+
+  /**
+   * Compare this.registryUrl against npm configs and write registry entry "registry" entry to <packageDir>/.npmrc
+   * if either is not equal
+   *
+   * @param packageDirectory
+   */
+  public async setNpmRegistry(packageDirectory: string): Promise<void> {
+    if (this.registryEntryLocal !== this.registryUrl || this.registryEntryGlobal !== this.registryUrl) {
+      let npmrc: string[] = await this.readNpmrc(packageDirectory);
```

Comment:
`string[]` here is redundant since it's implied from `readNpmrc`

Reply:
👍